src/GAP/iso_oscar_gap.jl

```diff
@@ -383,8 +383,11 @@ julia> preimage(f, y) == pol
 true
 ```
 """
-@attr Map function iso_oscar_gap(F)
-   return _iso_oscar_gap(F)
+@attr Map function iso_oscar_gap(R)
+  iso = _iso_oscar_gap(R)
+  S = codomain(iso)
+  GAP.Globals.SetIsoGapOscar(S, inv(iso))
```

Comment:
As @ThomasBreuer reminded me: not that this line may not do anything if `IsoGapOscar` is already set to a different value. So we may want to be paranoid and do something like this instead:
```suggestion
  f = inv(iso)
  GAP.Globals.SetIsoGapOscar(S, f)
  @assert GAP.Globals.IsoGapOscar(S) === f  # could differ if IsoGapOscar was set before
```